### ISSUE TITLE
Bug in AutoMPO with fermionic operators on same site (?)

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensors.jl#514]
I noticed a discrepancy when comparing ITensors.jl to C++ ITensor for 4-body electron-electron interaction terms.

This kind of expression turns up when using momentum space DMRG, where the Hubbard model interactions end up as terms like:
```julia
ampo += "Cdagdn", i, "Cdagup", j, "Cdn", k, "Cup", l
```
Here, I want that expression to work generically, even when `i == k`, `j == l`, etc. If I leave the interaction terms in the form above, the Julia version gives the wrong energy (when checked against the real space result, for example). When I replace those cases explicitly with interaction terms written in terms of `"Nup"` or `"Ndn"`, ITensors.jl matches with C++ ITensor (and the real space result).

I may be thinking about the fermion sign wrong here, but the replacement comes down to replacing terms like:
```julia
ampo += "Cdagdn", 1, "Cdagup", 1, "Cdn", 1, "Cup", 1
```
with:
```julia
ampo += "Ndn", 1, "Nup", 1
```
Basically, when I do replacements like that in the Julia code, the momentum space Hubbard model calculation matches with both the equivalent momentum space C++ calculation and the equivalent real space Hubbard model calculation. However, looking at that expression, I wouldn't expect those two terms to be equivalent from the fermion exchange rules, so there is definitely still some confusion (like maybe Julia's AutoMPO is doing the correct thing, but the transformation to momentum space was incorrect, and somehow the C++ ITensor is accidentally getting the answer right in the end).

I need to dig into the details more to find where the discrepancy is and come up with a minimal example, but I wanted to write this up before I forgot what issue I was seeing.

**Comment by mtfishman** [comment]
I'll double check that #519 fixes this issue (I presume it will).

**Comment by emstoudenmire** [comment]
Yes that would be a very good idea to check. I did run the unit tests including a new one plus the fermionic AutoMPO tests and the Hubbard model DMRG test. But since this bug appeared in the first place it shows that our tests certainly have some holes in them!
